atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleNamespaceDeleterIntegrationTest.java

```diff
@@ -0,0 +1,483 @@
+
+        Function<TableReference, OracleDdlTable> ddlTableFactory = tableReference -> OracleDdlTable.create(
+                tableReference,
+                connectionSupplier,
+                ddlConfig,
+                oracleTableNameGetter,
+                new TableValueStyleCache(),
+                executorService);
+
+        return new OracleNamespaceDeleter(OracleNamespaceDeleterParameters.builder()
+                .tablePrefix(ddlConfig.tablePrefix())
+                .overflowTablePrefix(ddlConfig.overflowTablePrefix())
+                .connectionSupplier(connectionSupplier)
+                .tableNameGetter(oracleTableNameGetter)
+                .oracleDdlTableFactory(ddlTableFactory)
+                .userId(connectionConfig.getDbLogin())
+                .build());
+    }
+
+    @After
+    public void after() {
+        namespaceDeleter.deleteAllDataFromNamespace();
+        namespaceDeleterWithNonDefaultPrefix.deleteAllDataFromNamespace();
+        namespaceDeleterWithDefaultPrefixNoMapping.deleteAllDataFromNamespace();
+        executorService.shutdown();
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDropsAllTablesAndOverflowTablesWithConfigPrefix() {
+        createTwoLogicalTablesWithDefaultPrefix();
+
+        namespaceDeleter.deleteAllDataFromNamespace();
+
+        assertSnapshotIsEmptyExcludingTimestampTable(getTableDetailsForDefaultPrefixes());
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDropsAllTablesAndOverflowTablesDespiteCaseInsensitiveMatchingNames() {
+        AllTableDetailsSnapshot kvsTablesSnapshot = getTableDetailsForDefaultPrefixes();
+
+        createTableAndOverflowTable(TABLE_NAME_ONE);
+        // Creating a table with the same name works in that we create the tables and the table mapping, but we fail
+        // when adding to the metadata table. We must still clear this up!
+        assertThatThrownBy(() -> createTableAndOverflowTable(TABLE_NAME_ONE_CASE_INSENSITIVE_MATCH));
+        AllTableDetailsSnapshot newTables =
+                AllTableDetailsSnapshot.difference(getTableDetailsForDefaultPrefixes(), kvsTablesSnapshot);
+        assertThat(newTables.physicalTableNames()).hasSize(4);
+        assertThat(newTables.prefixedTableNames()).hasSize(4);
+        assertThat(newTables.tableReferences()).hasSize(1);
+
+        namespaceDeleter.deleteAllDataFromNamespace();
+
+        assertSnapshotIsEmptyExcludingTimestampTable(getTableDetailsForDefaultPrefixes());
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDoesNotThrowWhenNoTablesToDelete() {
+        // Oracle throws on an empty IN () clause, whereas sqlite does not.
+        namespaceDeleter.deleteAllDataFromNamespace();
+        assertSnapshotIsEmptyExcludingTimestampTable(getTableDetailsForDefaultPrefixes());
+
+        assertThatNoException().isThrownBy(namespaceDeleter::deleteAllDataFromNamespace);
```

Comment:
nit: Is fine, though convention is normally `assertThatCode(namespaceDeleter::deleteAllDataFromNamespace).doesNotThrowAnyException();`